pkg/cloudprovider/providers/gce/gce_healthchecks.go

```diff
@@ -59,8 +52,88 @@ func (gce *GCECloud) CreateHttpHealthCheck(hc *compute.HttpHealthCheck) error {
 	return gce.waitForGlobalOp(op)
 }
 
-// ListHttpHealthCheck lists all HttpHealthChecks in the project.
+// ListHttpHealthChecks lists all HttpHealthChecks in the project.
 func (gce *GCECloud) ListHttpHealthChecks() (*compute.HttpHealthCheckList, error) {
 	// TODO: use PageToken to list all not just the first 500
 	return gce.service.HttpHealthChecks.List(gce.projectID).Do()
 }
+
+// Legacy HTTPS Health Checks
+
+// GetHttpsHealthCheck returns the given HttpsHealthCheck by name.
+func (gce *GCECloud) GetHttpsHealthCheck(name string) (*compute.HttpsHealthCheck, error) {
```

Comment:
Hmm, only the GCP documentation refers to these health checks as legacy. The network load balancer still requires HttpHealthCheck for probing. I'm not sure if adding "Legacy" is necessary at this point (but will be some day). We could add "(Legacy)" in each func comment.